packages/core/src/components/graphs/area-stacked.ts

```diff
@@ -0,0 +1,186 @@
+// Internal Imports
+import { Component } from "../component";
+import * as Configuration from "../../configuration";
+import { Roles, ScaleTypes, Events, TooltipTypes } from "../../interfaces";
+
+// D3 Imports
+import { area } from "d3-shape";
+import { select, color } from "d3";
+
+export class StackedArea extends Component {
+	type = "area-stacked";
+
+	areaGenerator: any;
+
+	// TODORF - Remove these listeners in destroy()
+	init() {
+		// Highlight correct area on legend item hovers
+		this.services.events.addEventListener("legend-item-onhover", e => {
+			const { hoveredElement } = e.detail;
+
+			this.parent
+				.selectAll("g.areas")
+				.transition(
+					this.services.transitions.getTransition("legend-hover-area")
+				)
+				.attr("opacity", d => {
+					if (d[0].group !== hoveredElement.datum().name) {
+						return Configuration.areas.opacity.unselected;
+					}
+
+					return Configuration.areas.opacity.selected;
+				});
+		});
+
+		// Un-highlight areas on legend item mouseouts
+		this.services.events.addEventListener("legend-item-onmouseout", e => {
+			this.parent
+				.selectAll("g.areas")
+				.transition(
+					this.services.transitions.getTransition(
+						"legend-mouseout-area"
+					)
+				)
```

Comment:
```suggestion
				.transition(this.services.transitions.getTransition("legend-mouseout-area"))
```

Reply:
code formatting related comments are somewhat irrelevant now since prettier will format them back to be that way again